python/lsst/ts/MTDome/encoding_tools.py

```diff
@@ -57,4 +756,60 @@ def decode(st):
     -------
         A decoded Python representation of the string.
     """
-    return json.loads(st)
+    data = json.loads(st)
+    validate(data)
+    return data
+
+
+def validate(data):
+    """Validates the data against a JSON schema and logs an error in case the
+    validation fails.
+
+    There are eight schemas: one for the commands, one for each of six status
+    command responses and one for all other command responses. This function
+    determines which schema to use based on the keys in the data. Commands are
+    validated as well to ensure that the simulator receives correct commands
+    and this should be done by other clients too.
+
+    Parameters
+    ----------
+    data: `dict`
+        The data to validate. The format of the dict is explained in the
+        `encode` function.
+
+    Raises
+    ------
+    ValidationError:
+        In case the validation fails.
+    ValueError:
+        In case the retrieved data doesn't contain a known key.
+
+    """
+    try:
+        if "command" in data.keys():
+            jsonschema.validate(data, command_schema)
+        elif "timeout" in data.keys():
+            jsonschema.validate(data, response_schema)
+        elif "AMCS" in data.keys():
+            jsonschema.validate(data, amcs_status_schema)
+        elif "ApSCS" in data.keys():
+            jsonschema.validate(data, apscs_status_schema)
+        elif "LCS" in data.keys():
+            jsonschema.validate(data, lcs_status_schema)
+        elif "LWSCS" in data.keys():
+            jsonschema.validate(data, lwscs_status_schema)
+        elif "MonCS" in data.keys():
+            jsonschema.validate(data, moncs_status_schema)
+        elif "ThCS" in data.keys():
+            jsonschema.validate(data, thcs_status_schema)
+        else:
+            message = f"Validation failed because no known key found in data {data!r}"
+            if validation_raises_exception:
+                raise KeyError(message)
```

Comment:
KeyError is a bit misleading. I understand the logic, but KeyError is kind-of reserved for dict. Better use RuntimeError(message), so if true KeyError happens, you will not get it..

Reply:
Thanks, I modified that.